packages/portal/document-viewer/src/components/DocumentViewer.vue

```diff
@@ -14,6 +25,12 @@ const props = defineProps({
   },
 })
 
+const { i18n } = composables.useI18n()
+
+const hasRequiredProps = computed((): boolean => {
+  return !!props.document
+})
```

Comment:
question: if the `document` prop is required but not provided, this should already be showing a Vue warning in the console. 

Is the purpose of the `document-viewer-error` UI for a deployed host app (e.g. Konnect running in Production) or for the developer working locally?

Reply:
True, the `document-viewer-error` UI is to surface it gracefully to the end user for a deployed host app.